### DISCUSSION TITLE
query regarding docker and conda?

### DISCUSSION THREAD

**Discussion by innat** [open-edge-platform/anomalib#619] [Q&A] [answered]
1. Why pick `devel` as base image when it looks not necessary and image file would be large? why not pick `runtime` or `base`?
2. Instead of conda, what do you think about [poetry](https://python-poetry.org/)?

**Comment by ashwinvaidya17** [comment, marked as answer]
@innat we use `devel` as it contains build tools needed for `pre-commit` and `tox` (and probably package publishing). It is convenient compared to manually adding the required packages and pinning their versions. 
The docker file uses `conda` just of convenience as it is easier to install the right python version and setup the virtual environment. Our `setup.py` is independent of any virtual environment tools. But we will have a look at poetry.
